backend/tests/documents/test_upload_document.py

```diff
@@ -0,0 +1,80 @@
+import base64
+import uuid
+from unittest import mock
+
+import pytest
+
+from src import context
+from src.documents.upload_document import (
+    decode_file_content,
+    generate_secure_filename,
+    upload_file_data,
+    upload_file_to_cloud,
+)
+from src.storage import CloudStorage
+
+context = context.ApplicationContext()
+
+
+def setup_function():
+    context.reset()
+
+
+# Cleanup
+def teardown_function():
+    context.reset()
+
+
+def test_uploading_file_data_returns_document_id():
+    mock_file_content = b"Hello, this is a test file."
+    decoded_file_content = base64.b64encode(mock_file_content).decode("utf-8")
+
+    mock_cloud_storage = mock.MagicMock()
+    mock_cloud_storage.put_object.return_value = None
+    context.register(CloudStorage, mock_cloud_storage)
+
+    response = upload_file_data("original.txt", decoded_file_content, "mock_bucket", "mock_folder")
+
+    assert response is not None
+    assert isinstance(response, str)
+
+
+def test_generating_secure_filename_works():
+    mock_original_filename = "Original.txt"
+
+    expected_secure_file_name, expected_document_id = generate_secure_filename(mock_original_filename)
```

Comment:
These aren't "expected", these are "actual" from actually calling the function under test.

Reply:
Ah, good catch!

Reply:
Fixed!